topics/icrc_34_get_delegation.md

```diff
@@ -0,0 +1,192 @@
+[![Extension Badge](https://img.shields.io/badge/EXTENDS-ICRC--25-ffcc222.svg)](./icrc_25_signer_interaction_standard.md)
+[![Standard Issue](https://img.shields.io/badge/ISSUE-ICRC--34-blue?logo=github)](https://github.com/dfinity/wg-identity-authentication/issues/115)
+
+<!-- TOC -->
+* [ICRC-34: Get Delegation](#icrc-34-get-delegation)
+  * [Summary](#summary)
+  * [Method](#method)
+  * [Scope (according to the ICRC-25 standard)](#scope-according-to-the-icrc-25-standard)
+    * [Example RPC Permission Request](#example-rpc-permission-request)
+  * [Requests](#requests)
+    * [Example RPC Get Delegation Request](#example-rpc-get-delegation-request)
+    * [Example RPC Get Delegation Response (wallet delegation)](#example-rpc-get-delegation-response-wallet-delegation)
+    * [Example RPC Get Delegation Response (app-specific delegation)](#example-rpc-get-delegation-response-app-specific-delegation)
+  * [Message Processing](#message-processing)
+  * [Errors](#errors)
+<!-- TOC -->
+
+## Summary
+
+When a relying party wants to remove wallet approval prompts and sign on behalf of its users, it uses the
+below `icrc34_get_delegation` method to obtain a delegation chain.
+
+The delegation chain's identifier may the user's wallet address (Wallet Delegation) or an address meant to be exclusively used 
+for the relying party (App-Specific Delegation).
+
+Wallet Delegations **MUST** be restricted by the signer to canister targets that trust the relying party following the 
+[ICRC-28](./icrc_28_trusted_origins.md) standard. Doing otherwise would allow malicious actors to steal user assets.
+
+App-Specific Delegations **MUST** be restricted by the signer to individual relying parties.
+
+Signers **SHOULD** give users the choice to authenticate with their wallets (if available) or with an app-specific identifier.
+
+Relying parties **SHOULD** remove targets if they want to isolate users from the rest of the ecosystem.
+
+## Method
+
+**Name and Scope:** `icrc34_get_delegation`
+
+**Prerequisite:** ICRC-25 granted permission scope `icrc34_get_delegation` or `*`.
+
+## Scope (according to the [ICRC-25 standard](./icrc_25_signer_interaction_standard.md))
+
+**Scope:** `icrc34_get_delegation`
+
+**Required Properties:**
+
+- `publicKey` (`text` string): An Ed25519 or ECDSA session key as described [here](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures).
+
+**Optional Properties:**
+
+- `targets` (`text` array): A list of target canister ids (textual representation) the scope is restricted to.
+- `maxTimeToLive` (`text` array): Expiration of the delegation in nanoseconds, though signer can still choose to
+return a delegation with a shorter expiration.
+- `idleOptions`
+
+### Example RPC Permission Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrc25_request_permissions",
+  "params": {
+    "scopes": [
+      {
+        "method": "icrc34_get_delegation"
+      }
+    ]
+  }
+}
+```
+
+## Requests
```

Comment:
I would structure this the same as e.g. ICRC-49: Have the schema / properties defined in this section, and then the example doesn't need to repeat that but can just be the JSON RPC request.

Reply:
updated